Assets/Decontainment/Singletons/BotManager.cs

```diff
@@ -71,4 +75,14 @@ public int GetTargetHeading(Controller targeter, int targetIndex)
         Vector2 look = target.transform.position - targeter.transform.position;
         return (int)Vector2.SignedAngle(targeter.transform.right, look);
     }
+
+    public IEnumerator PropagationCoroutine (int registerNumber, int registerValue, int botTeamID) {
```

Comment:
I highly-highly recommend making the coroutine method private and making a `public void PropagateRegister` method that sets up the coroutine for you. That way you don't have to type as much code to do the propagation like you did in the VirtualMachine class.